crates/biome_js_analyze/src/lint/nursery/no_useless_escape_in_string.rs

```diff
@@ -0,0 +1,180 @@
+    /// const s = "\n";
+    /// ```
+    ///
+    /// Tagged string template are ignored:
+    ///
+    /// ```js
+    /// const s = tagged`\a`;
+    /// ```
+    ///
+    /// JSX strings are ignored:
+    ///
+    /// ```jsx
+    /// <div attr="str\a"/>;
+    /// ```
+    ///
+    pub NoUselessEscapeInString {
+        version: "next",
+        name: "noUselessEscapeInString",
+        language: "js",
+        recommended: true,
+        severity: Severity::Error,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+impl Rule for NoUselessEscapeInString {
+    type Query = Ast<AnyString>;
+    type State = (JsSyntaxToken, usize);
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        match node {
+            AnyString::JsStringLiteralExpression(literal) => {
+                let tokrn = literal.value_token().ok()?;
+                let text = tokrn.text();
+                next_useless_escape(text, text.bytes().next()?).map(|index| (tokrn, index))
+            }
+            AnyString::JsTemplateExpression(template) => {
+                if template.tag().is_some() {
+                    return None;
+                }
+                for element in template.elements() {
+                    match element {
+                        AnyJsTemplateElement::JsTemplateChunkElement(chunk) => {
+                            let Ok(chunk) = chunk.template_chunk_token() else {
+                                continue;
+                            };
+                            if let Some(index) = next_useless_escape(chunk.text(), b'`') {
+                                return Some((chunk, index));
+                            }
+                        }
+                        AnyJsTemplateElement::JsTemplateElement(_) => {}
+                    }
+                }
+                None
+            }
+            AnyString::JsLiteralMemberName(member_name) => {
+                let Ok(token) = member_name.value() else {
+                    return None;
+                };
+                if token.kind() == JsSyntaxKind::JS_STRING_LITERAL {
+                    let text = token.text_trimmed();
+                    next_useless_escape(text, text.bytes().next()?).map(|index| (token, index))
+                } else {
+                    None
+                }
+            }
+        }
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, (token, index): &Self::State) -> Option<RuleDiagnostic> {
+        let escape_start = token
+            .text_trimmed_range()
+            .start()
+            .checked_add((*index as u32 + 1).into())?;
+        let escaped_char = token.text()[(1 + index)..].chars().next()?;
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            TextRange::at(escape_start, (escaped_char.len_utf8() as u32).into()),
+            markup! {
+                "The character doesn't need to be escaped."
```

Comment:
The diagnostics miss the second pillar of the rule pillars: explain the error, in this case we explain why the doesn't require escaping